### PR TITLE
save analytics params from request header into d1

### PR DESCRIPTION
**Pull request by KamaniBhavin**
1. Incorporate essential request parameters, such as bundle details and connecting IP, into a D1 database to enhance analytical capabilities.
2. Integrate the migration command into the pipeline to ensure smooth deployment.

### CODE REVIEW

**Comment by twodayslate** [comment]
Can we log every x-gateway header besides the API key one? Can we also add the URL path (host already in the header).

**Comment by KamaniBhavin** [comment]
> Can we log every x-gateway header besides the API key one? Can we also add the URL path (host already in the header). Not sure if you can just store a JSON dump of the url request itself?
> 
> Other than that looks good to me.

ok, will make the required changes if feasible and add test cases for it

File: migrations/0000_create_requests_table.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+-- Migration number: 0000 	 2023-12-20T02:07:03.645Z
+
+DROP TABLE IF EXISTS requests;
```
Comment (twodayslate) [inline migrations/0000_create_requests_table.sql:3]: Is this run every time we deploy? I don't want to lose data on deployments

Comment (KamaniBhavin) [reply migrations/0000_create_requests_table.sql:3]: no, it will only execute once.

**Comment by twodayslate** [comment]
lgtm

**Comment by twodayslate** [comment]
Lgtm

**Pull request merged**